src/drivers/distance_sensor/lightware_sf45_serial/lightware_sf45_serial.cpp

```diff
@@ -164,35 +164,73 @@ int SF45LaserSerial::collect()
 	/* read from the sensor (uart buffer) */
 	const hrt_abstime timestamp_sample = hrt_absolute_time();
 
-	if (_sensor_state == 1) {
+	if (_consecutive_fail_count > 50 && !_sensor_ready) {
```

Comment:
Question: When the state machine is restarted, `_consecutive_fail_count` will not be resetted to 0, but will stay at 50.

For `_consecutive_fail_count` to be resetted to 0 it is actually required, that `collect()` returns true. But as soon as `_consecutive_fail_count` is above 50 (and `!_sensor_ready`) `collect()` will directly cause restarting the state machine. And thus we end up here again. How is this loop exited?

Reply:
That is correct, the only would would be to init to break out of the loop (but this won't work if the sensor baud rate is misconfigured and can't get to the streaming state).

I fixed this with a restart of consecutive fail count if over the error threshold when collect() returns PX4_ERROR